packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -81,9 +88,22 @@ export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
         }
     }
 
-    public shouldRunLastSummary(): boolean {
+    public shouldRunLastSummary(stopReason: SummarizerStopReason): boolean {
+        if (stopReason === "parentNotConnected") {
```

Comment:
I think this should be !== 
Would be great to add tests for this (if possible).